src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+    program = f2008parser(reader)
+    fparser2specification = program.content[0].content[1].content[0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Character length specifications are not "
+            "supported.") in str(error.value)
+
+
+def test_fparser2astprocessor_process_declarations_intent():
+    '''Test that process_declarations method parses multiple specifications
+    of the intent attribute.
+    '''
+    from fparser.two.parser import ParserFactory
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    f2008parser = ParserFactory().create(std="f2008")
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    reader = FortranStringReader("integer, intent(in) :: arg1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg1").name == "arg1"
+    assert fake_parent.symbol_table.lookup("arg1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg1").shape == []
+    assert fake_parent.symbol_table.lookup("arg1").access == 'read_arg'
+
+    reader = FortranStringReader("integer, intent( IN ) :: arg2")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg2").name == "arg2"
+    assert fake_parent.symbol_table.lookup("arg2").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg2").shape == []
+    assert fake_parent.symbol_table.lookup("arg2").access == 'read_arg'
+
+    reader = FortranStringReader("integer, intent( Out ) :: arg3")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg3").name == "arg3"
+    assert fake_parent.symbol_table.lookup("arg3").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg3").shape == []
+    assert fake_parent.symbol_table.lookup("arg3").access == 'write_arg'
+
+    reader = FortranStringReader("integer, intent ( InOut ) :: arg4")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg4").name == "arg4"
+    assert fake_parent.symbol_table.lookup("arg4").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg4").shape == []
+    assert fake_parent.symbol_table.lookup("arg4").access == 'readwrite_arg'
+
+
+def test_fparser2astprocessor_process_declarations_array_attributes():
+    '''Test that process_declarations method parses multiple specifications
```

Comment:
"handles various different specifications of array attributes."?

Reply:
Updated, but with s/array/variable/